staging/src/k8s.io/apimachinery/pkg/apis/meta/fuzzer/fuzzer.go

```diff
@@ -248,32 +247,23 @@ func v1FuzzerFuncs(codecs runtimeserializer.CodecFactory) []interface{} {
 				metav1.LabelSelectorOpDoesNotExist,
 			}
 
-			if j.MatchExpressions != nil {
-				// NB: the label selector parser code sorts match expressions by key, and sorts the values,
-				// so we need to make sure ours are sorted as well here to preserve round-trip comparison.
-				// In practice, not sorting doesn't hurt anything...
-
-				for i := range j.MatchExpressions {
-					req := metav1.LabelSelectorRequirement{}
-					c.Fuzz(&req)
-					req.Key = randomLabelKey(c)
-					req.Operator = validOperators[c.Rand.Intn(len(validOperators))]
-					if req.Operator == metav1.LabelSelectorOpIn || req.Operator == metav1.LabelSelectorOpNotIn {
-						if len(req.Values) == 0 {
-							// we must have some values here, so randomly choose a short length
-							req.Values = make([]string, c.Rand.Intn(2)+1)
-						}
-						for i := range req.Values {
-							req.Values[i] = randomLabelPart(c, true)
-						}
-						sort.Strings(req.Values)
```

Comment:
why are we *dropping* the sorting in this fuzzer function?

if we added deduplication here and left the sorting in place, would this do everything we needed?

Reply:
mm... just read the PR description

> That function has been updated to no longer do any sorting or deduplication, because in general, roundtrips of the structured metav1.LabelSelector should preserve order and duplicates

Are we sure the staging/src/k8s.io/client-go/scale/roundtrip_test.go is the only place we're round-tripping fuzzed LabelSelector objects to the string form?

Given we have no "internal" LabelSelector type and there's no conversion for it (they are all straight copies), I'm not actually worried at all that `metav1.LabelSelector <-> metav1.LabelSelector` will not faithfully round-trip.

This fuzz function is widely used, making it strictly better at producing LabelSelector objects that will round-trip to the string form seems more helpful, and makes this PR ~tiny:

```diff
-sort.Strings(req.Values)
+req.Values = sets.List(sets.New(req.Values...)) // sort and deduplicate
```

Reply:
> Are we sure the staging/src/k8s.io/client-go/scale/roundtrip_test.go is the only place we're round-tripping fuzzed LabelSelector objects to the string form?

I don't see any other conversions that do this, but metav1.LabelSelector is converted to labels.Selector in so many places across the codebase (inside and outside of tests) that I can't say with confidence that nothing else operates on a fuzzed metav1.LabelSelector and assumes that the struct/string conversions are lossless.

I removed the scale-specific fuzz function and made the existing function deduplicate.